### PR TITLE
Migration doc

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: migration/1.3.x-1.4.x/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+files, in 1.3 it was possible to store very large files and parallelize
+directly over them. While this was convenient for a number of cases it
+wound up iteracting in confusing ways with the filesystem layouts we use
+for data processing and with our incrementality features. It also felt
+a little too magical because it abstracted key details of how computations
+would be parallelized away from the user. In 1.4 files can still be large
+but large files can only be processed by a single process. If data needs
+to be processed in parallel it must be spit into multiple smaller files,
+to help with this syntax has been added to the API to help split files.
+
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: Very nice explanation of the reasoning behind making files as the smallest processing units.

File: migration/1.3.x-1.4.x/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+getting both clusters running, if you're running them in the same
+Kubernetes cluster namespaces can be useful for avoiding name conflicts.
+Once both clusters are up and running you'll need to find an address for
+your 1.3 Pachyderm service so you can give it to the 1.4 cluster. This
+should be readily via `kubectl get all --namespace=<1.3 namespace>` Then
+to put data from a 1.3 repo into a 1.4 repo do:
+
+```
+pachctl put-file <1.4 repo> <1.4 branch> -f <1.3 ip address>:30650/<1.3 repo>/<1.3 branch> -r
+```
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: This command only migrates the head of the branch, right?  Should we make that clear?  Should we talk about how one might migrate all commits, i.e. preserving commit history?

File: migration/1.3.x-1.4.x/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Notice the `-r` parameter which tells Pachyderm to recursively scan the
+contents of the branch rather than just putting a single file, you can
+also migrate one file at a time by specifying it after the branch
+parameter in the address.
+
+Note that now is the time that you need to think about how you want to
+parallelize over your data. If your repo contains large files which were
+processed in 1.3 with `BLOCK` or `MAP` pipelines then you should use the
+`--split` flag in your `put-file` invocation to split the data up now.
+Refer to the [`put-file` docs](PLACEHOLDER) to learn more about `--split` works.
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: Did you want to use a real link here?

File: migration/1.3.x-1.4.x/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+After data has been migrated you can safely turn down the 1.3 cluster.
+
+## Pipelines
+
+### Inputs
+
+By far the biggest change to the Pipeline Spec is to the `inputs` field. Inputs
+no longer have a `method` field, instead they specify how their inputs can be
+parallelized using a glob pattern. [Read more about glob patterns
+here](PLACEHOLDER).
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: Same here.  Should there be a real link?

File: migration/1.3.x-1.4.x/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+than the file `foo` then you'll want to use `/*/*/*`.
+
+In 1.3 it wasn't possible to do `FILE` partitioning below the root layer
+of directories, in 1.4 this is possible with glob patterns. I.e. in the
+above example where a file is put to `dir/foo` with `--split` you could
+process all the pieces of the file together by passing `/*/*`.
+
+1.4 currently does not have a notion of incrementality, much of the
+incremental functionality present in 1.3 currently happens automatically.
+We'll be introducing a more advanced form of incrementality in a later
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: Should we talk about how the "more advanced form of incrementality" is basically the equivalent of `/prev`?

File: migration/1.3.x-1.4.x/README.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+version of 1.4 for incremental workloads like sum operations.
+
+In 1.4 `input`s are targeted at specific branches rather than all branches
+in a repo, this field defaults to master if left blank.
```
Comment (derekchiang) [inline migration/1.3.x-1.4.x/README.md]: What field?  Should probably make it clear that it's the `branch` field.

File: migration/1.3.x-1.4.x/README.md (hunk 3 above)
Comment (JoeyZwicker) [inline migration/1.3.x-1.4.x/README.md]: Dan and I will add the links in another PR. We need to do a lot of linking once the structure is set.

**Comment by dwhitena** [comment]
@JoeyZwicker Actually our migration doc already points to this place generally (i.e., telling people where to look for specific migration instructions scripts).  However, it may be good to explicitly mention this one as an important migration to note.

**Pull request merged**